### ISSUE TITLE
Purpose of "Bind user accounts to domains"

### ISSUE THREAD

**Issue by JoePJisc** [MISP/MISP#6170]
### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Support
| OS version (server)       | CentOS
| OS version (client)       | Win10
| PHP version               | 7.2.27
| MISP version / git hash   | 2.4.129, c0bc1a0c67bffc91cfa7382bb5df3d6bb39a42cd (via RPM)
| Browser                   | Google Chrome

### Support Questions

Please can you clarify what should happen when a domain is populated into "Bind user accounts to domains" on the Organisation page, I cannot find any mention of in the [User Guide](https://www.circl.lu/doc/misp/) and do not see any noticable change after setting this up with each orgs expected email domains.

My expectation from the name was:

* Self-registration requests get automatically mapped to the organisation based on email domain, even if Organisation and UUID are left blank, and,
* Org Admins can review and approve / reject requests for their own org.

Currently, the following is the case:

* Self registrations without org or UUID show "Requested organisation not found"
* Org admins get permission denied for the pending registrations page

### Logs, screenshots, configuration dump, ...

![Organisation Page](https://user-images.githubusercontent.com/60189659/89393415-ad8fa480-d702-11ea-85e5-c2ed01fd28ef.png)

![Self Registration Page](https://user-images.githubusercontent.com/60189659/89392222-2988ed00-d701-11ea-92e3-e6c0c2af3bc8.png)

![Pending Registration Page](https://user-images.githubusercontent.com/60189659/89392321-4cb39c80-d701-11ea-86a0-1795d262e49c.png)

**Comment by iglocska** [comment]
The self-registration approvals are purely for site-admins currently. It was a little bit less ambiguous a few months ago what it meant since the domain lock pre-dates the self-registration system.

Basically the idea is this: Org admins can create users, by setting the domain restriction an org admin can only create new users for e-mail addresses that match the domain requirements. The use-case is for example giving access to an organisation to self-management and they start filling your user roster with @gmail.com / @hotmail.com - potentially abusing their right to create users for your community for 3rd parties.

**Comment by JoePJisc** [comment]
That's really useful thanks, it looks like the current implementation should meet our needs for now.

It would be good if self-registration could be integrated with this in the future.

**Comment by iglocska** [comment]
Yeah, indeed, let's keep this open for now, it would indeed be great if org admins could create users based on registrations.
